### PR TITLE
Fix tiling inst seg OOM due to large amount of ground-truth masks

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

* Due to cases where there are more than 10,000 gt masks, limit the number of ground truths by randomly selecting 5000 masks 

### Checklist

- [x] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [x] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [x] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [x] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: tests/unit/algorithms/detection/tiling/test_tiling_detection.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def create_otx_dataset(height: int, width: int, labels: List[str]):
     """
     labels = []
     for label in ["rectangle", "ellipse", "triangle"]:
-        labels.append(LabelEntity(name=label, domain=Domain.DETECTION))
+        labels.append(LabelEntity(name=label, domain=Domain))
```
Comment (goodsong81) [inline tests/unit/algorithms/detection/tiling/test_tiling_detection.py]:
```suggestion
        labels.append(LabelEntity(name=label, domain=domain))
```
Was this your intention?

File: tests/unit/algorithms/detection/tiling/test_tiling_detection.py [hunk 2]
```diff
@@ -314,3 +315,56 @@ def test_tile_ir_scale_deploy(self, tmp_dir_path, scale_factor):
...
+            )
+        )
+
+        dataset = build_dataset(train_data_cfg)
+        train_dataloader = build_dataloader(dataset, **dataloader_cfg)
+        for data in train_dataloader:
+            assert isinstance(data["img"].data[0], torch.Tensor)
+            assert isinstance(data["gt_bboxes"].data[0][0], torch.Tensor)
+            assert isinstance(data["gt_labels"].data[0][0], torch.Tensor)
+            assert isinstance(data["gt_masks"].data[0][0], torch.Tensor)
```
Comment (goodsong81) [inline tests/unit/algorithms/detection/tiling/test_tiling_detection.py]: Is this block is actually checking the `max_annotation` feature?

**Review by goodsong81 — DISMISSED** [review]
Thank you for the quick fix.
Just a few comments.

File: tests/unit/algorithms/detection/tiling/test_tiling_detection.py (hunk 1 above)
Comment (eugene123tw) [reply tests/unit/algorithms/detection/tiling/test_tiling_detection.py]: @goodsong81 yes! Thanks for the catch!

File: tests/unit/algorithms/detection/tiling/test_tiling_detection.py (hunk 2 above)
Comment (eugene123tw) [reply tests/unit/algorithms/detection/tiling/test_tiling_detection.py]: no, so I changed it a bit :+1:

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**